### PR TITLE
Xcode8 GM + swift3 migration

### PR DESCRIPTION
**Pull request by appsailor**

### CODE REVIEW

**Comment by jcavar** [comment]
Very nice. Thanks! I will try to review this as soon as possible.

File: PullToRefreshDemo/BeatAnimator.swift [hunk 1]
```diff
@@ -28,52 +28,52 @@ import QuartzCore
 
 class BeatAnimator: UIView, PullToRefreshViewDelegate {
     
-    private let layerLoader = CAShapeLayer()
-    private let layerSeparator = CAShapeLayer()
+    fileprivate let layerLoader = CAShapeLayer()
```
Comment (jcavar) [inline PullToRefreshDemo/BeatAnimator.swift]: Is fileprivate necessary here? We are not using this outside of this class?

File: PullToRefreshDemo/BeatAnimator.swift [hunk 2]
```diff
@@ -28,52 +28,52 @@ import QuartzCore
...
+    fileprivate let layerSeparator = CAShapeLayer()
```
Comment (jcavar) [inline PullToRefreshDemo/BeatAnimator.swift]: Same here, why fileprivate?

**Comment by appsailor** [comment]
That was part of the migration tool initiated changes. So kept them as
suggested. After all, these are just demo classes right?

File: Refresher/PullToRefreshView.swift [hunk 3]
```diff
@@ -121,62 +121,62 @@ public class PullToRefreshView: UIView {
...
-            scrollView.contentOffset = CGPointMake(scrollView.contentOffset.x, -insets.top)
+            scrollView.contentOffset = CGPoint(x: scrollView.contentOffset.x, y: -insets.top)
         }, completion: {finished in
             self.animator.pullToRefreshAnimationDidStart(self)
             self.action()
         })
     }
     
-    private func stopAnimating() {
+    fileprivate func stopAnimating() {
```
Comment (jcavar) [inline Refresher/PullToRefreshView.swift]: fileprivate/private again?

File: Refresher/PullToRefreshView.swift [hunk 4]
```diff
@@ -121,62 +121,62 @@ public class PullToRefreshView: UIView {
...
-            super.observeValueForKeyPath(keyPath, ofObject: object, change: change, context: context)
+            super.observeValue(forKeyPath: keyPath, of: object, change: change, context: context)
         }
     }
     
     
     //MARK: PullToRefreshView methods
 
-    private func startAnimating() {
+    fileprivate func startAnimating() {
```
Comment (jcavar) [inline Refresher/PullToRefreshView.swift]: fileprivate/private again?

File: Refresher/PullToRefreshView.swift [hunk 5]
```diff
@@ -29,28 +29,28 @@ private let ContentOffsetKeyPath = "contentOffset"
...
+    func pullToRefresh(_ view: PullToRefreshView, progressDidChange progress: CGFloat)
+    func pullToRefresh(_ view: PullToRefreshView, stateDidChange state: PullToRefreshViewState)
 }
 
-public class PullToRefreshView: UIView {
+open class PullToRefreshView: UIView {
     
-    private var scrollViewBouncesDefaultValue: Bool = false
-    private var scrollViewInsetsDefaultValue: UIEdgeInsets = UIEdgeInsetsZero
+    fileprivate var scrollViewBouncesDefaultValue: Bool = false
```
Comment (jcavar) [inline Refresher/PullToRefreshView.swift]: fileprivate/private again?

File: Refresher/PullToRefreshView.swift [hunk 6]
```diff
@@ -29,28 +29,28 @@ private let ContentOffsetKeyPath = "contentOffset"
...
+    fileprivate var scrollViewInsetsDefaultValue: UIEdgeInsets = UIEdgeInsets.zero
```
Comment (jcavar) [inline Refresher/PullToRefreshView.swift]: fileprivate/private again?

File: Refresher/PullToRefreshView.swift [hunk 7]
```diff
@@ -29,28 +29,28 @@ private let ContentOffsetKeyPath = "contentOffset"
...
 
-    private var animator: PullToRefreshViewDelegate
-    private var action: (() -> ()) = {}
+    fileprivate var animator: PullToRefreshViewDelegate
```
Comment (jcavar) [inline Refresher/PullToRefreshView.swift]: fileprivate/private again?

File: Refresher/PullToRefreshView.swift [hunk 8]
```diff
@@ -29,28 +29,28 @@ private let ContentOffsetKeyPath = "contentOffset"
...
+    fileprivate var action: (() -> ()) = {}
```
Comment (jcavar) [inline Refresher/PullToRefreshView.swift]: fileprivate/private again?

File: Refresher/PullToRefreshView.swift [hunk 9]
```diff
@@ -29,28 +29,28 @@ private let ContentOffsetKeyPath = "contentOffset"
...
 
-    private var previousOffset: CGFloat = 0
+    fileprivate var previousOffset: CGFloat = 0
```
Comment (jcavar) [inline Refresher/PullToRefreshView.swift]: fileprivate/private again?

File: Refresher/Animator.swift [hunk 10]
```diff
@@ -27,29 +27,29 @@ import UIKit
...
-    private let titleLabel: UILabel = {
+    fileprivate let titleLabel: UILabel = {
         let label = UILabel()
         label.translatesAutoresizingMaskIntoConstraints = false
         return label
     }()
     
-    private let activityIndicatorView: UIActivityIndicatorView = {
-        let activity = UIActivityIndicatorView(activityIndicatorStyle: .Gray)
+    fileprivate let activityIndicatorView: UIActivityIndicatorView = {
```
Comment (jcavar) [inline Refresher/Animator.swift]: fileprivate/private again?

File: Refresher.podspec [hunk 11]
```diff
@@ -1,29 +1,28 @@
...
-  s.license      = "MIT"
+  s.license      = 'MIT'
 
-  s.author             = { "Josip Cavar" => "josipcavar04@gmail.com" }
-  s.social_media_url   = "http://twitter.com/josip04"
+  s.author             = { 'Josip Cavar' => 'josipcavar04@gmail.com' }
+  s.social_media_url   = 'http://twitter.com/josip04'
 
-  s.platform     = :ios, "8.0"
+  s.ios.deployment_target = '9.0'
```
Comment (jcavar) [inline Refresher.podspec]: Why was this needed?

**Comment by appsailor** [comment]
In case we are supporting other platforms in the future, the suggested
approach is to use deployment_target.

see
https://guides.cocoapods.org/syntax/podspec.html#group_multi_platform_support

As opposed to the platform attribute, the deployment_target attribute
allows to specify multiple platforms on which this pod is supported —
specifying a different deployment target for each.

File: Refresher.podspec (hunk 11 above)
Comment (jcavar) [inline Refresher.podspec]: Yes, but should it be 8.0?

File: PullToRefreshDemo/BeatAnimator.swift (hunk 2 above)
Comment (jcavar) [inline PullToRefreshDemo/BeatAnimator.swift]: These are, but some others aren't. It is simple change I think so I would prefer it to have right.
